packages/core/primitives/event-dispatch/src/event.ts

```diff
@@ -13,22 +13,24 @@ import {KeyCode} from './key_code';
 /**
  * Gets a browser event type, if it would differ from the JSAction event type.
  */
-export function getBrowserEventType(eventType: string) {
+export function getBrowserEventType(eventType: string): string {
   // Mouseenter and mouseleave events are not handled directly because they
   // are not available everywhere. In browsers where they are available, they
   // don't bubble and aren't visible at the container boundary. Instead, we
   // synthesize the mouseenter and mouseleave events from mouseover and
   // mouseout events, respectively. Cf. eventcontract.js.
-  if (eventType === EventType.MOUSEENTER) {
-    return EventType.MOUSEOVER;
-  } else if (eventType === EventType.MOUSELEAVE) {
-    return EventType.MOUSEOUT;
-  } else if (eventType === EventType.POINTERENTER) {
-    return EventType.POINTEROVER;
-  } else if (eventType === EventType.POINTERLEAVE) {
-    return EventType.POINTEROUT;
+  switch (eventType) {
+    case EventType.MOUSEENTER:
+      return EventType.MOUSEOVER;
+    case EventType.MOUSELEAVE:
+      return EventType.MOUSEOUT;
+    case EventType.POINTERENTER:
+      return EventType.POINTEROVER;
+    case EventType.POINTERLEAVE:
+      return EventType.POINTEROUT;
+    default:
+      return eventType;
```

Comment:
Suggestion: instead of the if-else and the switch approaches why don't you go with a dictionary for object lookup? also it's to me better for maintenance (extension) and readability

```suggestion
const eventMapping = {
  [EventType.MOUSEENTER]: EventType.MOUSEOVER,
  [EventType.MOUSELEAVE]: EventType.MOUSEOUT,
  [EventType.POINTERENTER]: EventType.POINTEROVER,
  [EventType.POINTERLEAVE]: EventType.POINTEROUT,
};
return eventMapping[eventType] || eventType;
```

Reply:
I made some tests on jsperf.app and jsbench.me with these three solutions (if, switch and dictionary).
The dictionary had good perfs but only if you define the `eventMapping` outside the function. In this case it could be at the end of the file with the other const, otherwise it may be up to 90% slower than the previous approach. What do you think @JeanMeche?

Reply:
Yes I wouldn't go with a map, as it introduces a lookup.

Reply:
Doing EventType.MOUSEENTER is also a lookup, since enums aren't actually a thing in JS and as far as I'm aware, enum values aren't inlined in the final bundle.

So the current code does actually do more lookups than than the object-map approach, since if you precompile the lookup-table as suggested, you are always doing two lookups, while the old code does always do at least two lookups and can do up to 5.

Reply:
Actually `EventType` isn't a typescript enum but a pure key value js object.
Does the lookup issue you're talking about still accurate in this case?